sonar-db/src/main/java/org/sonar/db/measure/MeasureQuery.java

```diff
@@ -92,17 +87,38 @@ public boolean returnsEmpty() {
       || (metricKeys != null && metricKeys.isEmpty());
   }
 
+  @Override
+  public boolean equals(@Nullable Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    MeasureQuery that = (MeasureQuery) o;
+    return Objects.equals(analysisUuid, that.analysisUuid) &&
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1067)